source/MaterialXTest/RenderGLSL.cpp

```diff
@@ -0,0 +1,432 @@
+    {
+    }
+
+  protected:
+    const std::string& languageTargetString() override
+    {
+        return _languageTargetString;
+    }
+    
+    bool runTest(const RenderUtil::RenderTestOptions& testOptions) const override
+    {
+        return (testOptions.languageAndTargets.count(_languageTargetString) > 0);
+    }
+
+    void loadLibraries(mx::DocumentPtr dependLib,
+        RenderUtil::RenderTestOptions& options) override;
+
+    void createShaderGenerator() override
+    {
+        _shaderGenerator = mx::GlslShaderGenerator::create();
+    }
+    void registerLights(mx::DocumentPtr dependLib, 
+        const RenderUtil::RenderTestOptions &options, mx::GenContext& context) override;
+
+    void createValidator(std::ostream& log) override;
+    bool runValidator(
+        const std::string& shaderName,
+        mx::TypedElementPtr element,
+        mx::GenContext& context,
+        mx::DocumentPtr doc,
+        std::ostream& log,
+        const RenderUtil::RenderTestOptions& testOptions,
+        RenderUtil::RenderProfileTimes& profileTimes,
+        const mx::FileSearchPath& imageSearchPath,
+        const std::string& outputPath = ".") override;
+
+    void getImplementationWhiteList(mx::StringSet& whiteList) override;
+
+    std::string _languageTargetString;
+    mx::GlslValidatorPtr _validator;
+    mx::LightHandlerPtr _lightHandler;
+};
+
+void GlslShaderRenderTester::loadLibraries(mx::DocumentPtr dependLib, 
+                                           RenderUtil::RenderTestOptions& options)
+{
+    // Read in light libraries and light rig to use
+    mx::FilePath lightDir = mx::FilePath::getCurrentPath() / mx::FilePath("resources/Materials/TestSuite/Utilities/Lights");
+    if (options.lightFiles.size() == 0)
+    {
+        GenShaderUtil::loadLibrary(lightDir / mx::FilePath("lightcompoundtest.mtlx"), dependLib);
+        GenShaderUtil::loadLibrary(lightDir / mx::FilePath("lightcompoundtest_ng.mtlx"), dependLib);
+        GenShaderUtil::loadLibrary(lightDir / mx::FilePath("light_rig.mtlx"), dependLib);
+    }
+    else
+    {
+        for (auto lightFile : options.lightFiles)
+        {
+            GenShaderUtil::loadLibrary(lightDir / mx::FilePath(lightFile), dependLib);
+        }
+    }
+
+}
+
+// Set light handler on validator based on loaded in lights
+void GlslShaderRenderTester::registerLights(mx::DocumentPtr dependLib, 
+    const RenderUtil::RenderTestOptions &options, mx::GenContext& context)
+{
+    //RenderUtil::AdditiveScopedTimer glslSetupLightingTimer(profileTimes.languageTimes.setupTime, "GLSL setup lighting time");
+    // Add lights as a dependency
+    _lightHandler = mx::LightHandler::create();
+    RenderUtil::createLightRig(dependLib, *_lightHandler, context,
+        options.radianceIBLPath, options.irradianceIBLPath);
+}
+
+
+//
+// Create a validator with an image and geometry handler
+// If a filename is supplied then a stock geometry of that name will be used if it can be loaded.
```

Comment:
This comment: "If a filename is supplied..." is that still valid? I don't see a filename being supplied anywhere.